Board.java

```diff
@@ -30,33 +28,123 @@ public Board playerMakesMove(char player, int position) {
         return this;
     }
 
+    public String getWinner() {
+        for (String combination : allWinningCombinations()) {
```

Comment:
combination is a line, and this would read more clearly if variable was called 'line'.
e.g. if a line contains all X's then X is the winner, if O then O is the winner else return Empty, or '-'.
Now might be a good time to think about your 'magic' values: x, o and -.  
What if you wanted to have different values?  You'd have to change these all over the place...